### PR TITLE
Allow offset without limit

### PR DESCRIPTION
**Pull request by mik629**
* let the `offset` be specified without a `limit`
* let the `limit` be specified without an `offset`
* let `offset` and `limit` be specified together via separate routines

### CODE REVIEW

File: ktorm-core/src/main/kotlin/org/ktorm/dsl/Query.kt [hunk 1]
```diff
@@ -330,24 +329,53 @@ public fun ColumnDeclaring<*>.desc(): OrderByExpression {
...
+    if (isOffsetInvalid && isLimitInvalid) {
         return this
     }
-
     return this.withExpression(
-        when (expression) {
-            is SelectExpression -> expression.copy(offset = offset, limit = limit)
-            is UnionExpression -> expression.copy(offset = offset, limit = limit)
-        }
+            when (expression) {
```
Comment (vincentlauvlwj) [inline ktorm-core/src/main/kotlin/org/ktorm/dsl/Query.kt:369]: We'd prefer 4-space indent here, but there are 8.

File: ktorm-support-mysql/src/test/kotlin/org/ktorm/support/mysql/MySqlTest.kt [hunk 2]
```diff
@@ -26,6 +26,16 @@ import java.util.concurrent.TimeoutException
...
     companion object {
+        const val TOTAL_RECORDS = 4
+        const val MINUS_ONE = -1
+        const val ZERO = 0
+        const val ONE = 1
+        const val TWO = 2
+        const val ID_1 = 1
+        const val ID_2 = 2
+        const val ID_3 = 3
+        const val ID_4 = 4
```
Comment (vincentlauvlwj) [inline ktorm-support-mysql/src/test/kotlin/org/ktorm/support/mysql/MySqlTest.kt:37]: We don't have to define these numbers as constants. It makes no sense. Seems the quality check doesn't ignore the test code?

**Review by vincentlauvlwj — CHANGES_REQUESTED** [review]
Thanks for your contribution! I will merge your code in the next version. Please also update the `build.gradle` file, add your GitHub ID to the developers info (line 90).

File: ktorm-support-mysql/src/test/kotlin/org/ktorm/support/mysql/MySqlTest.kt (hunk 2 above)
Comment (mik629) [reply ktorm-support-mysql/src/test/kotlin/org/ktorm/support/mysql/MySqlTest.kt:37]: Yep, it doesn't

File: ktorm-core/src/main/kotlin/org/ktorm/dsl/Query.kt (hunk 1 above)
Comment (mik629) [reply ktorm-core/src/main/kotlin/org/ktorm/dsl/Query.kt]: Done

**Comment by mik629** [comment]
Could you please mark it with `hacktoberfest-accepted` label? 🙏

**Comment by vincentlauvlwj** [comment]
OK

**Pull request merged**